core/src/main/java/io/questdb/std/str/Utf8String.java

```diff
@@ -0,0 +1,157 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.std.str;
+
+import io.questdb.std.Unsafe;
+import org.jetbrains.annotations.NotNull;
+
+import java.lang.reflect.Field;
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.stream.IntStream;
+
+/**
+ * An immutable UTF-8 string over owned native memory, originating from a Java String.
+ */
+public class Utf8String implements Utf8NativeCharSequence {
```

Comment:
maybe we can avoid introducing this object, let's discuss

Reply:
Discussed in person. Keeping it as whilst the approach in this review causes Gc it is the lesser evil.
The new string type is to be used solely by the `TableToken` type.

The impl is now renamed to `GcUtf8String` and paired down to its minimal implementation.